python/ray/train/tests/test_base_trainer.py

```diff
@@ -57,7 +57,10 @@ def check_cpus(self):
 
     assert ray.available_resources()["CPU"] == 4
     trainer = DummyTrainer(
-        check_cpus, scaling_config=ScalingConfig(trainer_resources={"CPU": 2})
+        check_cpus,
+        scaling_config=ScalingConfig(
+            trainer_resources={"CPU": 2}, resources_per_worker={}
+        ),
```

Comment:
This is because we have 1 worker with 1 cpu, so the available resources would be 1 if we didn't set this to empty resource dict?

Reply:
Yes. By default `resources_per_worker = {"CPU": 1}`, and we will use trainer (2 x CPU) + 1 default worker (1 x CPU) = 3 CPUs in total.